### PR TITLE
feat(pipelines): add support for collapsible groups

### LINKED ISSUE #153: Support for collapsible groups in topology pipelines
**Describe the problem**
The topology pipelines components and layout should handle collapsible groups. This would include:

- Adding a decorator on the group label to allow collapsing similar to standard topology group nodes
- Displaying collapses groups to look like a TaskNode but similar adjustments as in DefaultGroupCollapsed
- Layout should adjust when groups are collapsed/expanded  

**What is your product and what release date are you targeting?**
RHOAI: sprint 2.10 (March 25 - April 2).

### PR DESCRIPTION
**Pull request by jenny-s51**
## What

Will close #153.

WIP / help wanted: 
- ~~Recursive selection of subgroups when parent group is selected~~ 
☑️ Discussed with design, selection to be replaced with expand action animation to indicate expanded state

- ~~Draw location and edge anchor points of collapsed pipeline group~~
☑️ Fixed by @jeff-phillips-18 

## Description

Adds support for collapsible pipelines groups and related styling as discussed with @andrew-ronaldson and @jeff-phillips-18  

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Surge preview:
https://expand-collapse.surge.sh/

### CODE REVIEW

**Comment by andrew-ronaldson** [comment]
I'll pull this down and take a look. Thanks for tagging me.
We'll need a v6 branch update to take in this new functionality. I have a branch open now where I am applying Penta tokens.

File: packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    if (element.isCollapsed()) {
+      return (
+        <PipelinesDefaultGroupCollapsed
+          className={className}
+          element={element}
+          onCollapseChange={handleCollapse}
+          badge={`${childCount}`}
+          badgeColor="#f5f5f5"
+          badgeBorderColor="#d2d2d2"
+          badgeTextColor="#000000"
```
Comment (jeff-phillips-18) [inline packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx:109]: Use styles here so they can be overridden.

File: packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    }
+    return (
+      <PipelinesDefaultGroupExpanded
+        className={className}
+        labelPosition={LabelPosition.top}
+        element={element}
+        onCollapseChange={handleCollapse}
+        badgeColor="#f5f5f5"
+        badgeBorderColor="#d2d2d2"
+        badgeTextColor="#000000"
```
Comment (jeff-phillips-18) [inline packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx:122]: Use styles here so they can be overridden.

File: packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx (hunk 1 above)
Comment (jenny-s51) [reply packages/module/src/pipelines/components/groups/PipelinesDefaultGroup.tsx]: Nice catch, thank you @jeff-phillips-18 ! Let me know if these changes are what you had in mind.

**Review by jeff-phillips-18 — APPROVED** [review]

**Review by jessiehuff — APPROVED** [review]

**Review by andrew-ronaldson — APPROVED** [review]
LGTM, thanks!

**Pull request merged**